components/CredentialSwitch.vue

```diff
@@ -0,0 +1,85 @@
+<template>
+  <component
+    :is="selection"
+    v-bind="{...$props, ...$attrs}">
+    <template
+      v-for="(_, slot) in $slots"
+      #[slot]="scope">
+      <slot
+        :name="slot"
+        v-bind="scope || {}" />
+    </template>
+  </component>
+</template>
+
+<script setup>
+import {computed, defineProps, unref} from 'vue';
+import {config} from '@bedrock/web';
+import CredentialBase from './CredentialBase.vue';
+import CredentialDetail from './CredentialDetail.vue';
+import {merge} from 'lodash-es';
+
+const componentDefaults = {
+  list: CredentialBase,
+  details: CredentialDetail
+};
+
+const props = defineProps({
+  mode: {
+    type: String,
+    default: 'list'
+  },
+  credential: {
+    type: Object,
+    required: true
+  }
+});
+
+const selection = computed(() => {
+  const {credentialDisplay: {registration: registrations}} = config;
+  const map = {};
```

Comment:
Use a `Map` instance instead. It's more performant and doesn't result in creating N-many hidden compiled classes in v8 (N = number of properties added).